### DISCUSSION TITLE
Is it possible to use fsReadStream with pipe?

### DISCUSSION THREAD

**Discussion by LeoMouyna** [Claviz/xlstream#43] [Q&A] [answered]
He there! 👋 

First of all thank you for this library!

I wanted to know if it's possible to use an opened fileReadStream with `pipe` to parse it ?
If you want more context, I'm downloading xlsx files from s3 as stream and I want to process decoded lines on the flow.

For now I think I have to download and save files locally then use `getXlsxStream` with local path then once process is done I delete local file.

Am I missing something?

**Comment by jansivans** [comment, marked as answer]
Hello!
Unfortunately, with `xlstream`, it is not possible to parse files "on the fly." 
The only available approach is to work with locally downloaded files, as you have already been doing.
